sdk/core/Azure.Core/src/Page.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace Azure
+{
+    /// <summary>
+    /// A single <see cref="Page{T}"/> of values from a request that may return
+    /// zero or more <see cref="Page{T}"/>s of values.
+    /// </summary>
+    /// <typeparam name="T">The type of values.</typeparam>
+    public readonly struct Page<T>
+    {
+        /// <summary>
+        /// Gets the values in this <see cref="Page{T}"/>.
+        /// </summary>
+        public IReadOnlyList<T> Values { get; }
```

Comment:
Open question - we've got open issues like #6894 to use `T[]` (rather than `IEnumerable<T>` or `IReadOnlyList<T>`) and it'd be great to be consistent.